Add the write permission for pull requests to the snapshot action.

dlorenc opened the pull request:
This failed in https://github.com/sigstore/root-signing/actions/runs/1913427789/attempts/1
because of the default permissions change we made.

Signed-off-by: Dan Lorenc <lorenc.d@gmail.com>

<!--
Thanks for opening a pull request!

Please remember to:
- mention any issue(s) that this PR closes using a closing keyword as well as the issue number, such as "Closes #XYZ" or "Resolves sigstore/repo-name#XYZ", cf.
  [documentation](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- ensure your commits are signed-off, as sigstore uses the [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin) using `git commit -s`, or `git commit -s --amend` if you want to amend already existing commits
- lastly, ensure there are no merge commits!
Thank you :)
-->

#### Summary
<!--
A description of what this pull request does, as well as QA test steps (if applicable and if not already added to the Jira ticket).
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/sigstore/YYYYYY/issues/XXXXX

-->
Fixes

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates)
* API additions—new endpoint, new response fields, or newly accepted request parameters
* Database changes (any)
* Websocket additions or changes
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating)
* New features and improvements, including behavioural changes, UI changes and CLI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense.

-->
```release-note

```

SantiagoTorres commented on cmd/verify/main.go:
```diff
@@ -167,22 +173,117 @@ func main() {
+			repo, err := tuf.NewRepo(local)
+			if err != nil {
+				log.Printf("error reading trusted TUF local: %s", *tufRoot)
+				return
```
Not sure of the context, but would it be useful to return Error/a negative integer so as to make calling processes (and shells) aware that things failed?

asraa replied:
Yeah, the places where it really errors I exit with 1. I should do that here!
Thank you!

asraa replied:
done!

asraa commented:
would love re-approval for manual branch protection :)

dlorenc committed:
Add the write permission for pull requests to the snapshot action.

This failed in https://github.com/sigstore/root-signing/actions/runs/1913427789/attempts/1
because of the default permissions change we made.

Signed-off-by: Dan Lorenc <lorenc.d@gmail.com>
